### PR TITLE
Cover legacy sendonly using {offerToReceiveAudio/Video: false}.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fix for https://github.com/w3c/webrtc-pc/issues/1461.

***
[Preview](https://s3.amazonaws.com/pr-preview/jan-ivar/webrtc-pc/offertoreceivefalse.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/webrtc-pc/cc8d80f...jan-ivar:eef2557.html)

### CODE REVIEW

**Comment by fippo** [comment]
Are people using this to put someone on hold and we need to cover the recvonly->inactive case too?

File: webrtc.html [hunk 1]
```diff
@@ -3484,7 +3484,13 @@
                   prior to the createOffer() call.
                 </p>
                 <p>
-                  In all other situations, it will be disregarded.
+                  Whenever this is given a false value, and the
+                  <code><a>RTCPeerConnection</a></code> has a non-stopped
+                  "sendrecv" audio transceiver, createOffer() MUST as its first
+                  step set that transceiver's
+                  [[<a href="#direction-slot">Direction</a>]] slot
```
Comment (taylor-b) [inline webrtc.html]: This slot link can be updated.

File: webrtc.html [hunk 2]
```diff
@@ -3484,7 +3484,13 @@
                   prior to the createOffer() call.
                 </p>
                 <p>
-                  In all other situations, it will be disregarded.
+                  Whenever this is given a false value, and the
+                  <code><a>RTCPeerConnection</a></code> has a non-stopped
+                  "sendrecv" audio transceiver, createOffer() MUST as its first
```
Comment (taylor-b) [inline webrtc.html]: Sorry to be pedantic, but what if there are multiple such transceivers? Does this affect all of them, just the first, just the last?

**Comment by jan-ivar** [comment]
@fippo hmm that [actually works](https://jsfiddle.net/jib1/c48q11jd/), though is anyone is using it? It'd only work on one side.

File: webrtc.html (hunk 2 above)
Comment (jan-ivar) [inline webrtc.html]: Firefox [appears](https://jsfiddle.net/jib1/kvzj12sb/) to apply it to all.

**Comment by jan-ivar** [comment]
Went with all. @taylor-b PTAL.

**Comment by jan-ivar** [comment]
I also added @fippo's bit since it matches two implementations.

**Comment by burnburn** [comment]
Perhaps offerToReceiveX, as legacy behavior, really only needs to be supported for initial configurations and not as a control for later behavior.

**Comment by jan-ivar** [comment]
> applications are likely to do a normal createOffer() later to take the call "off of hold"

@taylor-b That [doesn't actually work today](https://jsfiddle.net/jib1/c48q11jd/3/), but `createOffer({{offerToReceiveAudio: true})` [does](https://jsfiddle.net/jib1/c48q11jd/4/).

**Comment by jan-ivar** [comment]
@aboba What problem requires list discussion? Is this not good to merge?

**Comment by taylor-b** [comment]
> @taylor-b That doesn't actually work today, but createOffer({{offerToReceiveAudio: true}) does.

Ok, then the problem will be that `createOffer({{offerToReceiveAudio: true})` doesn't change the direction to `sendrecv`; same point.

The thing that needs list discussion is whether we want the `offerToReceive` functionality to cover just the most common usage (generating an "m=" section that otherwise wouldn't be present), or if we want to cover the direction-changing use cases as well.

**Comment by jan-ivar** [comment]
@taylor-b I'm happy to cover only basic usage. What text changes would you like to see?

**Comment by jan-ivar** [comment]
To clarify, I consider *basic usage* to include `true` and `false` for initial configurations, as I think @burnburn suggests.

**Comment by taylor-b** [comment]
Let me clarify. There are two classes of use cases I see for `offerToReceive`:

1. The endpoint wants to offer to receive media, but doesn't have media to send. `offerToReceive` is the only way to force audio/video m= sections to be generated in this case.
2. The endpoint wants to manipulate the direction attribute of the "m=" section, because it's dealing with a SIP endpoint.

1 is something I'd expect almost _every_ sufficiently robust application to be using; if the endpoint doesn't have a webcam, it probably still wants to be able to receive video. In the call where we decided to add `offerToReceive` to the legacy section, this was the motivation.

2 is more corner casey, and we could tell these applications "use the transceiver API". Supporting use cases that involve SIP endpoints is why we introduced the transceiver API in the first place...

**Comment by jan-ivar** [comment]
I guess I imagined:

  3. The endpoint has media to send but never wants to receive media. `offerToReceiveAudio/Video: false` is the only way to force "sendonly" as opposed to "sendrecv".

Are there no sites using `{offerToReceiveAudio/Video: false}` to set "sendonly" on initial connection?

**Comment by taylor-b** [comment]
I'm sure those sites exist, but my assumption was that they'd be much less common. What's an example of how an application might use this?

**Comment by jan-ivar** [comment]
> 1. The endpoint wants to receive media, but doesn't have media to send.

> 3. The endpoint has media to send but doesn't want to receive media.

Isn't this two sides of the same coin, the difference being who the offerer is?

I assumed they had the same use-cases.

**Comment by taylor-b** [comment]
> Isn't this two sides of the same coin, the difference being who the offerer is?

Not necessarily. If "A" has media to send but "B" doesn't, the typical offer/answer from A->B will be "sendrecv"/"recvonly", and from B->A will be "recvonly"/"sendonly". In neither case would there be a "sendonly" offer.

**Comment by jan-ivar** [comment]
OK I'm all for less legacy to support!

**Pull request closed without merging**